paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/DataTypeDefaultVisitor.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.paimon.hive;
+
+import org.apache.paimon.shade.guava30.com.google.common.collect.Lists;
+import org.apache.paimon.types.ArrayType;
+import org.apache.paimon.types.DataField;
+import org.apache.paimon.types.DataType;
+import org.apache.paimon.types.MapType;
+import org.apache.paimon.types.RowType;
+
+import java.util.List;
+
+/**
+ * DataTypeDefaultVisitor to visitor the sourceInspector and writerInspector to get the
+ * FieldDeserializer.
+ */
+public abstract class DataTypeDefaultVisitor<P, R> {
```

Comment:
I mean using `org.apache.paimon.types.DataTypeDefaultVisitor.`

Reply:
DataTypeDefaultVisitor or DataTypeVisitor may not apply in this case,they are have only one parameter,in hive we need tree param to deserialize value

Reply:
Maybe return a custom class to contains tree param?